src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -173,7 +196,17 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     if (finishedWork.effectTag !== NoEffect) {
       const current = finishedWork.alternate;
       commitWork(current, finishedWork);
-      commitLifeCycles(current, finishedWork);
+      const trappedError = commitLifeCycles(current, finishedWork);
+      if (trappedError) {
+        allTrappedErrors = allTrappedErrors || [];
+        allTrappedErrors.push(trappedError);
+      }
+    }
+
+    // Now that the tree has been committed, we can handle errors.
+    if (allTrappedErrors) {
+      // TODO: handle multiple errors with distinct boundaries.
```

Comment:
A plausible strategy: When an error happens you can keep walking through the unmounting children and the effect list until you hit the `lastEffect` edge of the boundary. Then you can handle the error right there without keeping an array of them. Then you continue committing the rest.

However, then the tree wouldn't be fully committed in `unstable_handleError` which is probably bad. So the array strategy might be needed after all.